deps/http-parser/http_parser.c

```diff
@@ -1992,6 +1992,9 @@ http_parse_host(const char * buf, struct http_parser_url *u, int found_at) {
   const char *p;
   size_t buflen = u->field_data[UF_HOST].off + u->field_data[UF_HOST].len;
 
+  if (buflen > UINT16_MAX)
+    return 1;
```

Comment:
It's unfortunate that we're diverging from upstream http-parser. I think we should apply this for now, but try to upstream those changes and then re-sync our copy.